executor/join.go

```diff
@@ -500,16 +505,40 @@ func (e *NestedLoopApplyExec) fetchBigRow(goCtx goctx.Context) (Row, bool, error
 	}
 }
 
-// prepare runs the first time when 'Next' is called and it reads all data from the small table and stores
-// them in a slice.
+func (e *NestedLoopApplyExec) fetchBigChunkRow(goCtx goctx.Context) (*chunk.Row, bool, error) {
```

Comment:
we can change this function name to `fetchSelectedOuterRow` and:
1. only return the selected outer row.
2. emit the unmatched outer row to `e.resultChunk`.
3. use the returned bool value to indicate whether there exists more outer rows.